packages/expo-splash-screen/android/src/main/java/expo/modules/splashscreen/SplashScreenController.kt

```diff
@@ -0,0 +1,151 @@
+package expo.modules.splashscreen
+
+import android.app.Activity
+import android.os.Handler
+import android.view.View
+import android.view.ViewGroup
+import expo.modules.splashscreen.exceptions.NoContentViewException
+import java.lang.ref.WeakReference
+
+const val SEARCH_FOR_ROOT_VIEW_INTERVAL = 20L
+
+class SplashScreenController(
+  activity: Activity,
+  resizeMode: SplashScreenImageResizeMode,
+  private val rootViewClass: Class<*>,
+  splashScreenConfigurator: SplashScreenConfigurator
+) {
+  private val weakActivity = WeakReference(activity)
+  private val contentView: ViewGroup = activity.findViewById(android.R.id.content) ?: throw NoContentViewException()
+  private var splashScreenView: View = SplashScreenView(activity, resizeMode, splashScreenConfigurator)
+  private val handler = Handler()
+
+  private var rootViewState = RootViewState.NO_ROOT_VIEW
```

Comment:
From what I can see `rootViewState` isn't used anywhere outside of `handleRootView` method, moreover, it's only ever set (never read from). Could we remove it, please? 🙏